tonic-build/src/lib.rs

```diff
@@ -211,11 +213,46 @@ fn generate_attributes<'a>(
         .collect::<Vec<_>>()
 }
 
+fn format_service_name<T: Service>(package: &str, service: &T) -> String {
+    format!(
+        "{}{}{}",
+        package,
+        if package.is_empty() { "" } else { "." },
+        service.identifier()
+    )
+}
+
+fn format_method_name<T: Service>(
+    package: &str,
+    service: &T,
+    method: &<T as Service>::Method,
+) -> String {
+    format!(
+        "{}{}{}.{}",
+        package,
+        if package.is_empty() { "" } else { "." },
+        service.identifier(),
+        method.identifier()
+    )
```

Comment:
I was thinking just this code could be put into a fn the rest I think we could leave

Reply:
Is it like this? Should I not move format_method_name to the lib?